### PR TITLE
Feature add validation test

### LINKED ISSUE #919: The entrance of each module of karmada needs more rigorous verification and more reasonable output
**What happened**:
if we run karmada-shcheduler with wrong flag，The following situations will occur

```
/bin/karmada-scheduler --secure-port=sdfsdf
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc000010001, 0x0, 0x0, 0x0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:1026 +0xe6
k8s.io/klog/v2.(*loggingT).output(0x40574e0, 0x3, 0x0, 0x0, 0xc0004fe460, 0x1, 0x384aced, 0x7, 0x12, 0xd00)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:975 +0x313
k8s.io/klog/v2.(*loggingT).printDepth(0x40574e0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc0005a7e30, 0x1, 0x1)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:735 +0x2fd
k8s.io/klog/v2.(*loggingT).print(0x40574e0, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0xc0005a7e30, 0x1, 0x1)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:717 +0x95
k8s.io/klog/v2.Fatal(0xc0005a7e30, 0x1, 0x1)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:1494 +0x87
main.main()
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/cmd/scheduler/main.go:18 +0x19d

goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x40574e0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:1169 +0x73
created by k8s.io/klog/v2.init.0
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/klog/v2/klog.go:420 +0xf1

goroutine 100 [sleep]:
time.Sleep(0x6fc23ac00)
	/usr/local/go/src/runtime/time.go:193 +0xd2
sigs.k8s.io/controller-runtime/pkg/log.init.0.func1()
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:63 +0x34
created by sigs.k8s.io/controller-runtime/pkg/log.init.0
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:62 +0x35

goroutine 115 [select]:
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x332cf68, 0x33d61e0, 0xc000544bd0, 0x4086b01, 0xc0003800c0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:167 +0x17b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x332cf68, 0x12a05f200, 0x0, 0xc0006e8301, 0xc0003800c0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x9f
k8s.io/apimachinery/pkg/util/wait.Until(0x332cf68, 0x12a05f200, 0xc0003800c0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
k8s.io/apimachinery/pkg/util/wait.Forever(0x332cf68, 0x12a05f200)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:81 +0x45
created by k8s.io/component-base/logs.InitLogs
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/component-base/logs/logs.go:58 +0x85

goroutine 94 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:165 +0x9d
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x27
created by os/signal.Notify.func1.1
	/usr/local/go/src/os/signal/signal.go:151 +0x45

goroutine 117 [chan receive]:
k8s.io/apiserver/pkg/server.SetupSignalContext.func1(0xc00019edc0)
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apiserver/pkg/server/signal.go:48 +0x36
created by k8s.io/apiserver/pkg/server.SetupSignalContext
	/Users/zhangxiao/go/src/github.com/wawa0210/karmada/vendor/k8s.io/apiserver/pkg/server/signal.go:47 +0x147
```

The following can be improved through the output:

- Strict verification for important flags
- The exception stack here does not need to be output

**What you expected to happen**:

```
/bin/karmada-scheduler --secure-port=sdfsdf
Usage:
  karmada-scheduler [flags]
  karmada-scheduler [command]

Available Commands:
  help        Help about any command
  version     Print the version information.

Flags:
      --add_dir_header                           If true, adds the file directory to the header of the log messages
      --alsologtostderr                          log to standard error as well as files
      --bind-address string                      The IP address on which to listen for the --secure-port port. (default "0.0.0.0")
      --enable-scheduler-estimator               Enable calling cluster scheduler estimator for adjusting replicas.
      --failover                                 Reschedule on cluster failure.
  -h, --help                                     help for karmada-scheduler
      --kube-api-burst int                       Burst to use while talking with karmada-apiserver. Doesn't cover events and node heartbeat apis which rate limiting is controlled by a different set of flags. (default 60)
      --kube-api-qps float32                     QPS to use while talking with karmada-apiserver. Doesn't cover events and node heartbeat apis which rate limiting is controlled by a different set of flags. (default 40)
      --kubeconfig string                        Path to a KubeConfig. Only required if out-of-cluster.
      --leader-elect                             Enable leader election, which must be true when running multi instances. (default true)
      --leader-elect-resource-namespace string   The namespace of resource object that is used for locking during leader election. (default "karmada-system")
      --log-flush-frequency duration             Maximum number of seconds between log flushes (default 5s)
      --log_backtrace_at traceLocation           when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                           If non-empty, write log files in this directory
      --log_file string                          If non-empty, use this log file
      --log_file_max_size uint                   Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                              log to standard error instead of files (default true)
      --master string                            The address of the Kubernetes API server. Overrides any value in KubeConfig. Only required if out-of-cluster.
      --one_output                               If true, only write logs to their native severity level (vs also writing to each lower severity level)
      --scheduler-estimator-port int             The secure port on which to connect the accurate scheduler estimator. (default 10352)
      --scheduler-estimator-timeout duration     Specifies the timeout period of calling the scheduler estimator service. (default 3s)
      --secure-port int                          The secure port on which to serve HTTPS. (default 10351)
      --skip_headers                             If true, avoid header prefixes in the log messages
      --skip_log_headers                         If true, avoid headers when opening log files
      --stderrthreshold severity                 logs at or above this threshold go to stderr (default 2)
  -v, --v Level                                  number for the log level verbosity
      --vmodule moduleSpec                       comma-separated list of pattern=N settings for file-filtered logging

Use "karmada-scheduler [command] --help" for more information about a command.

```

The following modules all have the problems mentioned in this issue

 

- [x] agent #930
- [x] controller-manager @Charlie17Li  #960
- [ ] kubectl-karmada
- [x] scheduler-estimator @jameszhangyukun #962
- [x] webook  @feeltimeQ #952

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- Others:

### PR DESCRIPTION
**Pull request by Charlie17Li**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

add validation test for a controller - manager and refactor the validation of controller-manager

**Which issue(s) this PR fixes**:
Part of #919 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by gy95** [comment]
please rebase to upstream/master to fix FOSSA errors(fixed in #959).

**Comment by RainbowMango** [comment]
Hi @Charlie17Li 
Please don't use  `git merge`.

I guess you have added karmada repo as `upstream`, then you can do the rebase with:
```bash
git fetch upstream
git rebase upstream/master
git push origin feature-add-validation-test:feature-add-validation-test -f 
```

**Comment by Charlie17Li** [comment]
> Hi @Charlie17Li Please don't use `git merge`.
> 
> I guess you have added karmada repo as `upstream`, then you can do the rebase with:
> 
> ```shell
> git fetch upstream
> git rebase upstream/master
> git push origin feature-add-validation-test:feature-add-validation-test -f 
> ```

ok, I am gonna try it

**Comment by Charlie17Li** [comment]
> please rebase to upstream/master to fix FOSSA errors(fixed in #959).

thanks to remind me :)

File: cmd/controller-manager/app/options/validation.go [hunk 1]
```diff
@@ -1,84 +1,38 @@
 package options
 
 import (
-	"fmt"
-
 	"github.com/karmada-io/karmada/pkg/util"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
```
Comment (RainbowMango) [inline cmd/controller-manager/app/options/validation.go:6]: Could you please group the imports according to `std packages` --> `third-party packages` --> `karmada packages`?

```suggestion
	"k8s.io/apimachinery/pkg/util/validation/field"
	
	"github.com/karmada-io/karmada/pkg/util"
```

**Comment by RainbowMango** [comment]
Hi @Charlie17Li 
Can you help squash your commits to one? 
You can do that by `rebase`:
```
git rebase -i HEAD~3
```
Then, update the second and third commit from `pick` to `fixup`.
```bash
pick 816d1e3 add validation test for controller-manager and refactor the validation of controller-manager
fixup 8009ad9 modify the format to pass the staticcheck
fixup 926d26f adjust the import order
```

**Comment by Charlie17Li** [comment]
> Hi @Charlie17Li Can you help squash your commits to one? You can do that by `rebase`:
> 
> ```
> git rebase -i HEAD~3
> ```
> 
> Then, update the second and third commit from `pick` to `fixup`.
> 
> ```shell
> pick 816d1e3 add validation test for controller-manager and refactor the validation of controller-manager
> fixup 8009ad9 modify the format to pass the staticcheck
> fixup 926d26f adjust the import order
> ```

copy that.

**Comment by RainbowMango** [comment]
/lgtm
/approve

Thanks @Charlie17Li ~~

**Pull request merged**